Delta/Emulation/GameViewController.swift

```diff
@@ -1070,6 +1087,71 @@ extension GameViewController
             emulatorCore.rate = emulatorCore.deltaCore.supportedRates.lowerBound
         }
     }
+    
+    func performStatusBarAction()
+    {
+        let enabled = ExperimentalFeatures.shared.showStatusBar.isStatusBarEnabled
+        ExperimentalFeatures.shared.showStatusBar.isStatusBarEnabled = !enabled
+        
+        if let pauseView = self.pauseViewController
+        {
+            pauseView.dismiss()
+        }
+        
+        self.setNeedsStatusBarAppearanceUpdate()
+    }
+    
+    func performScreenshotAction()
+    {
+        guard let snapshot = self.emulatorCore?.videoManager.snapshot() else { return }
+        
+        let scale = ExperimentalFeatures.shared.gameScreenshots.scale != nil ? ExperimentalFeatures.shared.gameScreenshots.scale?.rawValue : 1.0
+        let size = CGSize(width: snapshot.size.width * scale!, height: snapshot.size.height * scale!)
+        
+        UIGraphicsBeginImageContextWithOptions(size, false, 1.0)
+        UIGraphicsGetCurrentContext()!.interpolationQuality = .none
+        
+        snapshot.draw(in: CGRect(origin: CGPoint.zero, size: size))
+        let scaledSnapshot = UIGraphicsGetImageFromCurrentImageContext()
+        
+        UIGraphicsEndImageContext()
+        
+        if ExperimentalFeatures.shared.gameScreenshots.saveToPhotos
+        {
+            UIImageWriteToSavedPhotosAlbum(scaledSnapshot!, nil, nil, nil)
```

Comment:
`UIImageWriteToSavedPhotosAlbum` is an even older API...can you switch over to `PhotoKit.framework`? Also, we need to handle user permission and ask for access to image library (if we haven't already) without blocking main thread.